applications: nrf5340_audio: Move controller (#10238)
nrfconnect/sdk-nrf: doc/nrf/libraries/bin/bt_ll_pcft/index.rst

.. _lib_bt_ll_pcft_readme:

LE Audio Controller for nRF5340 (Net core controller)
#####################################################

.. contents::
   :local:
   :depth: 2

> This link layer controller for the nRF5340 NET core, is used for LE Audio applications.

Reviewer: ```suggestion
This link layer controller, for the nRF5340 NET core, is used for LE Audio applications.
```

This controller is given as a binary (hex file) and is provided by Packetcraft.

Overview
********

> Communication between the APP and NET core is handled by the HCIRPMsg library.

Reviewer: ```suggestion
Communication between the APP and NET core is handled by the HCI_RPMsg library.
```

*************
The controller has two main methods of configuration:
 * Standard Host-Controller Interface (HCI) commands
 * Vendor-Specific commands (VSC).
   * These are listed in <ble_hci_vsc.h>

Usage
*****
The controller is flashed to the nRF5340 NET core. Configurations and calls are handled through standard HCI and VSC.

Samples using the library
*************************
The :ref:`nrf5340_audio` application uses the LE Audio Controller.

Files
*****
.. |net_core_hex_note| replace:: The network core for both gateway and headsets are programmed with the precompiled Bluetooth Low Energy Controller binary file :file:`ble5-ctr-rpmsg_<X>.hex`, where *<X>* corresponds to the controller version, for example :file:`ble5-ctr-rpmsg_3216.hex`.
   This file includes the LE Audio Controller for nRF53 and is provided in the :file:`/bin` directory.
   If :ref:`DFU is enabled <nrf53_audio_app_configuration_configure_fota>`, the binary file will be generated in the :file:`build/zephyr/` directory and will be called :file:`net_core_app_signed.hex`.

There are two other files in the :file:`/bin` directory:
:file:`ble5-ctr-rpmsg_shifted_<X>.hex` which has been shifted to make space for mcuboot.

> :file:`ble5-ctr-rpmsg_shifted_min_<X>.hex`which is the same as above, but made for minimum mcuboot.

Reviewer: ```suggestion
:file:`ble5-ctr-rpmsg_shifted_min_<X>.hex` which is the same as above, but made for minimum mcuboot.
```

See the :ref:`lib_bt_ll_pcft_api` for more information.

Usage
*****

This controller is provided as a HEX file and includes the LE Audio Controller Subsystem for nRF53, on license from Packetcraft.
Check the license file in the :file:`/bin` directory for usage conditions.

When using the file, the controller is programmed to the network core of nRF5340.
Configurations and calls are handled through standard HCI and VSC.

Samples using the library
*************************

The :ref:`nrf53_audio_app` application uses the controller.

Files
*****

The network core for both gateway and headsets of the :ref:`nrf53_audio_app` is programmed with the precompiled Bluetooth Low Energy Controller binary file :file:`ble5-ctr-rpmsg_<XYZ>.hex`, where *<XYZ>* corresponds to the controller version, for example :file:`ble5-ctr-rpmsg_3216.hex`.
This file includes the LE Audio Controller Subsystem for nRF53.
If :ref:`DFU is enabled <nrf53_audio_app_configuration_configure_fota>`, the subsystem's binary file will be generated in the :file:`build/zephyr/` directory and will be called :file:`net_core_app_signed.hex`.

There are two other files in the :file:`/bin` directory:

> * :file:`ble5-ctr-rpmsg_shifted_<XYZ>.hex` which has been shifted to make space for mcuboot.

Author: Done
